pandas/core/reshape.py

```diff
@@ -410,13 +413,24 @@ def unstack(obj, level, fill_value=None):
 
     if isinstance(obj, DataFrame):
         if isinstance(obj.index, MultiIndex):
-            return _unstack_frame(obj, level, fill_value=fill_value)
+            unstacked = _unstack_frame(obj, level, fill_value=fill_value)
+        else:
```

Comment:
hmm I think this should be done *inside* the ``.stack`` and ``_unstack_frame`` functions no?